doc/contributor/howto-guide-adding-generated-libraries.md

```diff
@@ -183,8 +183,19 @@ were written by a robot:
 The Cloud documentation links (`cloud.google.com/*/docs/*`) in these files are
 not always valid. Find the correct urls and update the links.
 
-Also, be sure to document each service's endpoint environment variable in
-`doc/main.dox`
+## Document the service's endpoint environment variable
+
+```shell
+lib="google/cloud/${library}"
+services=$(ls ${lib}/*_connection.h | xargs -I {} basename {} _connection.h)
+for service in ${services[@]}; do
+  env_var=$(grep -o "GOOGLE_CLOUD_CPP_.*_ENDPOINT" ${lib}/internal/${service}_option_defaults.cc); \
+  endpoint=$(grep -ro "\"[[:alnum:]]*\.googleapis\.com" ${lib}/internal/${service}_option_defaults.cc); \
+  connection=$(grep -ro "\~[[:alnum:]]*()" ${lib}/${service}_connection.h); \
+  text="- \`${env_var}=...\` changes the default endpoint\n  (${endpoint:1}) used by \`${connection:1:-2}\`."; \
+  sed -i "s/\(- \`GOOGLE_CLOUD_CPP_ENABLE_TRACING=rpc\`\)/${text}\n\n\1/" ${lib}/doc/main.dox; \
```

Comment:
Should we try to make this idempotent?

Reply:
I think that is a wise suggestion. It may make maintenance slightly easier, if for example one service is added to an already documented library.

I didn't go through and add those `<~-- inject -->` tags everywhere. We might want to do that later. Then we could (maybe) move this script out of the scaffolding instructions and just run it as part of checkers (like how we inject quickstart code into the README's)

Reply:
> I didn't go through and add those `<~-- inject -->` tags everywhere. We might want to do that later. ...

Sounds like a good idea to me.